converter/service/converter.py

```diff
@@ -0,0 +1,215 @@
+        accept = s.headers.get(header_accept)
+        decoded_url = query.get(param_url)
+
+        try:
+            json_response = handle(decoded_url, user_agent, accept)
+            s.send_response(HTTPStatus.OK)
+            content_header = gen_universe.format_universe_repo_content_type(
+                _get_repo_version(accept))
+            s.send_header(header_content_type, content_header)
+            s.end_headers()
+            s.wfile.write(json_response.encode())
+        except Exception as e:
+            s.send_error(HTTPStatus.BAD_REQUEST, str(e))
+
+
+def handle(decoded_url, user_agent, accept):
+    """Results in either an error or the requested json data
+
+    :param decoded_url: The url to be fetched from
+    :type decoded_url: str
+    :param user_agent: User-Agent header value
+    :type user_agent: str
+    :param accept: Accept header value
+    :return Requested json data
+    :rtype str (a valid json object)
+    """
+    logger.debug('Url : %s\n\tUser-Agent : %s\n\tAccept : %s',
+                 decoded_url, user_agent, accept)
+    repo_version = _get_repo_version(accept)
+    dcos_version = _get_dcos_version(user_agent)
+    logger.debug('Version %s\nDC/OS %s', repo_version, dcos_version)
+
+    req = Request(decoded_url)
+    req.add_header(header_user_agent, user_agent)
+    req.add_header(header_accept, accept)
+    try:
+        res = urlopen(req, timeout=MAX_TIMEOUT)
+        charset = res.info().get_param(param_charset) or default_charset
+        raw_data = res.read(MAX_BYTES+1)
+        if len(raw_data) > MAX_BYTES:
+            logging.info('%s response exceeded the size limit %d',
+                         decoded_url,
+                         len(raw_data))
+            raise ValueError(ErrorResponse.MAX_SIZE.to_msg())
+        packages = json.loads(raw_data.decode(charset)).get(json_key_packages)
+    except (HTTPError, URLError) as error:
+        logger.info("Request protocol error %s", decoded_url)
+        logger.exception(error)
+        raise error
+
+    return render_json(packages, dcos_version, repo_version)
+
+
+def render_json(packages, dcos_version, repo_version):
+    """Returns the json
+
+    :param packages: package dictionary
+    :type packages: dict
+    :param dcos_version: version of dcos
+    :type dcos_version: str
+    :param repo_version: version of universe repo
+    :type repo_version: str
+    :return filtered json data based on parameters
+    :rtype str
+    """
+    processed_packages = gen_universe.filter_and_downgrade_packages_by_version(
+        packages,
+        dcos_version
+    )
+    updated_json_data = json.dumps({json_key_packages: processed_packages})
+    errors = gen_universe.validate_repo_with_schema(
+        json.loads(updated_json_data),
+        repo_version
+    )
+    if len(errors) != 0:
+        logger.error(errors)
+        raise ValueError(ErrorResponse.VALIDATION_ERROR.to_msg(errors))
+    return updated_json_data
```

Comment:
This needs to be simplified. At a high-level this:

1. Filter dict: dict -> dict
1. Project dict by enclosing it in a dict with `package` property: dict -> dict
1. Encode as a string using JSON: dct -> str
1. Decode string as a JSON into a dict: str -> dict
1. Validate the generated dict: dict -> errors

I think we can get rid of 3. and 4. if we return a `dict` or just 4. if we want to return a `str`.